x/ecocredit/server/marketplace/query_sell_orders.go

```diff
@@ -0,0 +1,132 @@
+package marketplace
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/marketplace/v1"
+	"github.com/regen-network/regen-ledger/types/ormutil"
+	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+)
+
+// SellOrders queries all sell orders in state with optional pagination
+func (k Keeper) SellOrders(ctx context.Context, req *marketplace.QuerySellOrdersRequest) (*marketplace.QuerySellOrdersResponse, error) {
+	pg, err := ormutil.GogoPageReqToPulsarPageReq(req.Pagination)
+	if err != nil {
+		return nil, err
+	}
+
+	it, err := k.stateStore.SellOrderTable().List(ctx, api.SellOrderSellerIndexKey{}, ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
+	defer it.Close()
```

Comment:
```suggestion
```


We are already closing iterator in [ormlist.Paginate](https://github.com/regen-network/regen-ledger/blob/dcc5dc72ce40d75e45323663b949fcd3629c7b4c/orm/iterator.go#L146).